pkg/sources/gcs/gcs.go

```diff
@@ -273,7 +273,7 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk) err
 				ctx.Logger().V(1).Info("error setting start progress progress", "name", o.name, "error", err)
 				return
 			}
-			s.setProgress(ctx, o.name, persistableCache)
+			s.setProgress(ctx, o.md5, persistableCache)
```

Comment:
correct, this sets the value in the cache to the hash. But i updated the log message to still use the object name since the hash is not very useful there.